### PR TITLE
DROOLS-1798: Include java.lang.Number in project imports by default

### PR DESCRIPTION
**Pull request by jomarko**
Ensemble with:
- kiegroup/drools-wb#720
- kiegroup/kie-soup#17

### CODE REVIEW

File: kie-wb-common-services/kie-wb-common-services-backend/src/main/java/org/kie/workbench/common/services/backend/project/ProjectImportsServiceImpl.java [hunk 1]
```diff
@@ -100,7 +100,15 @@ public ProjectImports load(final Path path) {
                 saveProjectImports(path);
             }
             final String content = ioService.readAllString(Paths.convert(path));
-            return projectConfigurationContentHandler.toModel(content);
+            final ProjectImports projectImports = projectConfigurationContentHandler.toModel(content);
+
+            // java.lang.Number imported by default in new guided rule
+            // include it into project imports if not present already
+            final Import javaLangNumber = new Import(java.lang.Number.class.getName());
+            if (projectImports.getImports().getImports().stream().filter(anImport -> anImport != null).noneMatch(anImport -> anImport.equals(javaLangNumber))) {
```
Comment (triceo) [inline kie-wb-common-services/kie-wb-common-services-backend/src/main/java/org/kie/workbench/common/services/backend/project/ProjectImportsServiceImpl.java]: If you use `Objects.equals()`, you will be able to get rid of the `filter()`.

File: kie-wb-common-services/kie-wb-common-services-backend/src/main/java/org/kie/workbench/common/services/backend/project/ProjectImportsServiceImpl.java [hunk 2]
```diff
@@ -100,7 +100,15 @@ public ProjectImports load(final Path path) {
                 saveProjectImports(path);
             }
             final String content = ioService.readAllString(Paths.convert(path));
-            return projectConfigurationContentHandler.toModel(content);
+            final ProjectImports projectImports = projectConfigurationContentHandler.toModel(content);
+
+            // java.lang.Number imported by default in new guided rule
+            // include it into project imports if not present already
+            final Import javaLangNumber = new Import(java.lang.Number.class.getName());
```
Comment (triceo) [inline kie-wb-common-services/kie-wb-common-services-backend/src/main/java/org/kie/workbench/common/services/backend/project/ProjectImportsServiceImpl.java]: A suggestion: if the `Import(String name)` constructor had a friend (`Import(Class<?> cls)`), callers wouldn't need to repeat `.getName()` all the time. Also, it looks a bit better, saying "import this class", as opposed to "import class represented by this textual FQN".

File: kie-wb-common-services/kie-wb-common-services-backend/src/main/java/org/kie/workbench/common/services/backend/project/ProjectImportsServiceImpl.java (hunk 1 above)
Comment (jomarko) [reply kie-wb-common-services/kie-wb-common-services-backend/src/main/java/org/kie/workbench/common/services/backend/project/ProjectImportsServiceImpl.java]: Great! I will apply.

File: kie-wb-common-services/kie-wb-common-services-backend/src/main/java/org/kie/workbench/common/services/backend/project/ProjectImportsServiceImpl.java (hunk 2 above)
Comment (jomarko) [reply kie-wb-common-services/kie-wb-common-services-backend/src/main/java/org/kie/workbench/common/services/backend/project/ProjectImportsServiceImpl.java]: Ok, will add such constructor to class `Import`.

**Comment by jomarko** [comment]
@triceo ready for re-review. @manstis please have a look when you back from PTO.

**Pull request merged**